### PR TITLE
feat(test): add {AMDGPU_TARGETS} substitution and emu test categories

### PR DESCRIPTION
**Pull request by archana-ramalingam**
## Motivation
  - `test_categories.yaml` categories that build/run kernels need `--gpu-targets` matching the runtime architecture. 
  - Currently this is hardcoded per category (e.g. `--gpu-targets,gfx1250`), requiring a separate category for each arch. 
  - Adding `{AMDGPU_TARGETS}` substitution lets categories be arch-agnostic.
  - Adding emu test categories.

## Technical Details
  - Substitute `{AMDGPU_TARGETS}`  — `AMDGPU_TARGETS` carries concrete gfx targets (e.g. `gfx1250`)
  - Populate `fetch-gfx-targets: ["gfx1250"]` for `gfx125X`. No hardware tests triggered (`run_tests: False` unchanged)
  - Add `emu-fast` & `emu-full` to `VALID_TEST_CATEGORIES` so global `TEST_TYPE` doesn't error on ctest components
  - Add 5 unit tests for substitution logic (include, exclude, empty-drops-token, keeps-non-token, no-op)
  -When `AMDGPU_TARGETS` is empty, markers containing the token are dropped (prevents malformed marker expressions)
  - Fallback to `extract_gpu_arch(AMDGPU_FAMILIES)` for pipelines that only set `AMDGPU_FAMILIES`
  - Requires companion rocm-libraries [PR #10360](https://github.com/ROCm/rocm-libraries/pull/10360), this PR lands first.

## Test Plan
  - `python -m pytest build_tools/github_actions/tests/pytest_runner_test.py` — 36/36 passed (5 new)
  - Existing non-emu test categories unaffected

## Test Result
  -  36/36 unit tests passed

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/TheRock/blob/main/CONTRIBUTING.md.

## Risk level
  - None — additive change. Existing test categories unaffected.
  - `fetch-gfx-targets` change only populates `AMDGPU_TARGETS` env var (no hardware tests triggered).

JIRA ID : AIHPBLAS-3988

### CODE REVIEW

**Review by dileepr1 — APPROVED** [review]
Changes looks good.
It would be helpful to explicitly note what would be the expected behavior if AMDGPU_TARGETS is empty.

File: build_tools/github_actions/test_executable_scripts/pytest_runner.py [hunk 1]
```diff
@@ -113,13 +118,23 @@ def extract_gpu_arch(amdgpu_families):
...
     Build a pytest -m expression from a category's marker config plus the
     GPU-arch skip markers registered in pytest.ini (skip-gfxXXXX).
+
+    ``{AMDGPU_TARGETS}`` tokens in marker strings are substituted with the
+    *amdgpu_target* value (a concrete gfx target like ``gfx1250``) so
+    categories can be arch-agnostic.
     """
-    include = category_config.get("pytest_markers", []) or []
-    exclude = category_config.get("exclude_markers", []) or []
+    sub = amdgpu_target or ""
```
Comment (geomin12) [inline build_tools/github_actions/test_executable_scripts/pytest_runner.py]: better variable name pls

File: build_tools/github_actions/test_executable_scripts/test_runner.py [hunk 2]
```diff
@@ -38,6 +38,9 @@
     "ffm-standard",
     "ffm-comprehensive",
     "ffm-full",
+    # emulation categories (AM/rocjitsu)
+    "emu-quick",
+    "emu-full",
```
Comment (geomin12) [inline build_tools/github_actions/test_executable_scripts/test_runner.py]: irrelevant changes, not in scope. pls remove

**Review by geomin12 — COMMENTED** [review]
lgtm, two nits

File: build_tools/github_actions/test_executable_scripts/pytest_runner.py (hunk 1 above)
Comment (archana-ramalingam) [reply build_tools/github_actions/test_executable_scripts/pytest_runner.py]: updated

File: build_tools/github_actions/test_executable_scripts/test_runner.py (hunk 2 above)
Comment (archana-ramalingam) [reply build_tools/github_actions/test_executable_scripts/test_runner.py]: PR title was probably misleading, updated now. 
`emu-quick` / `emu-full` are the only consumers of `{AMDGPU_TARGETS}` substitution — without the `VALID_TEST_CATEGORIES` entries, substitution ships with nothing that can use it. 
The 5 new unit tests also reference emu marker patterns.

**Comment by archana-ramalingam** [comment]
> Changes looks good. It would be helpful to explicitly note what would be the expected behavior if AMDGPU_TARGETS is empty.

Added comment

**Review by geomin12 — APPROVED** [review]
